### ISSUE TITLE
[Linux] Physical mouse and invisible walls in cannon fodder in amiga core PUAE

### ISSUE THREAD

**Issue by ZzackKbin** [libretro/RetroArch#15249]
Hello 
I have problem with physical mouse for linux arch. In Cannon Fodder when screen is scrolling mouse cursor stop on invisible walls when use physical mouse but when I use Sony pad DS4 everything is ok.
https://youtu.be/M_CwEI6_2iE

- in windows and the same hardware (steamdeck) no problems with physical mouse
- in linux arch desktop and only run retroarch physical mouse stop on invisble walls when screen is scrolling

I tested retroarch with lr-puae with even closed steam application.
https://github.com/libretro/libretro-uae/issues/608

**Comment by sonninnos** [comment]
Same will happen with every game and every core that uses a mouse. Did you try some other input driver yet if available?

**Comment by ZzackKbin** [comment]
Now I testesd settlers pc and Dosbox pure has the same problem with invisible walls for physical mouse in linux arch .
I tried change in drivers>input sdl linuxraw x and no results

**Comment by ZzackKbin** [comment]
I installed standalone dosbox and physical mouse works ok in settlers. Looks like something broken in retroarch for linux.

**Comment by i30817** [comment]
This was fixed a month ago. It was because (in wayland desktops), mouse capture wasn't implemented, so in fullscreen the cursor was 'floating'. This also happens in retroarch built for x11 running through XWayland in a wayland desktop iirc.

Maybe try a updated version of retroarch (not the cores)?

**Comment by ZzackKbin** [comment]
In SteamOS (linuxarch):
https://i.imgur.com/HQ6Nhep.png
https://i.imgur.com/1kWVTXv.png

I reisntalled and no results :(

**Comment by i30817** [comment]
write the git version and build date you have on the main menu->information window.

To be clear, this was fixed in these 3 commits in order (there are more wayland fixes but they don't interact with the mouse):

https://github.com/libretro/RetroArch/pull/15103  merged march 20
https://github.com/libretro/RetroArch/pull/15114 merged march 21
https://github.com/libretro/RetroArch/pull/15128 merged march 24

if the git commit you have on your version of retroarch is older than these, or it's from a fork that hasn't updated/synced to upstream recently, there is your cause.

**Comment by ZzackKbin** [comment]
sys info: https://i.imgur.com/xEpbREV.png

**Comment by i30817** [comment]
That makes it sure. That commit is from 16 march 4 days before this started being fixed.

commit 6616b807edea5b56683e9fbca8b003b436c22b13 (tag: v1.15.0)
Author: github-actions <github-actions@github.com>
Date:   Thu Mar 16 00:13:09 2023 +0000

    Fetch translations from Crowdin

**Comment by i30817** [comment]
BTW. this bug literally existed for YEARS and people complained about it regularly, so it being fixed just as you noticed is actually kind of 'lucky'.

**Comment by ZzackKbin** [comment]
Maybe Emudeck  something broken.

**Comment by i30817** [comment]
No. I already told you, this is literally a bug in retroarch. You're just not using the latest code of retroarch. When that code becomes available to update, this problem will disappear. Have some patience and play some other games meanwhile, it shouldn't take more than a month, and probably a few days.

**Comment by ZzackKbin** [comment]
thx

**Comment by dyha89** [comment]
Hi
I'm still experiencing this issue having RetroArch 1.16.0.
(Also in Cannon Fodder in PUAE.)

**Comment by i30817** [comment]
This issue is for Wayland and was pretty comprehensively fixed, there is a similar issue for android, and (maybe) max caused by pretty much the same cut corner, but in other drivers. 

Is your device running Linux or Android or mac\ios?

https://github.com/libretro/RetroArch/issues/15016 (Mac issue)
https://github.com/libretro/RetroArch/issues/14016 (android issue)

**Comment by dyha89** [comment]
Thanks for your answer. I'm using Manjaro Linux. I have installed retroarch using pacman package manager.

**Comment by dyha89** [comment]
https://github.com/libretro/RetroArch/assets/25767123/37d1fdbd-73d8-4677-966d-5bbf6bb34066

**Comment by dyha89** [comment]
![image](https://github.com/libretro/RetroArch/assets/25767123/00f3e3d0-2336-4e82-9bac-26b41a169d36)

**Comment by sonninnos** [comment]
Core version, game and the mouse core option do not matter one bit, except of course there is no mouse at all if disabled. What matters is that the emulated mouse cursor can't travel at the same speed as the OS cursor, and isn't 1:1 at the same coordinates, which will result in out of bounds situations if mouse is not grabbed to the window properly. As in the core does not get new mouse readings when the OS mouse cursor is no longer inside the window.

AFAIK that was fixed ages ago in Linux, but perhaps then not with all input drivers. With Windows it hasn't been an issue for a long time regardless if playing in a window or not, or if mouse grab is active or not. Generally in fullscreen mouse grab should not even matter, but again not all input drivers behave the same.

With Amiga it is very easy to test the mouse cursor without launching any games. Just make sure you are launching Kickstart version 2 or later, and hold both mouse buttons down on startup to activate the Early Startup menu.

**Comment by dyha89** [comment]
I have noticed that 'Automatic Mouse Grab' in input options was off. I turned it on, and I had no reproduction after 15 minutes of playing.
![image](https://github.com/libretro/RetroArch/assets/25767123/7b41bd49-d2f8-4a12-a339-85bb9345ba80)
I'll let you know again if it still works fine after some more time.

**Comment by sonninnos** [comment]
Indeed mouse grab is only enforced on fullscreen if using non-windowed fullscreen. That option is for always  grabbing when window becomes active.

**Comment by i30817** [comment]
Seems like it should be forced on Fullscreen, full stop. Why isn't it?

**Comment by dyha89** [comment]
Since I turned Automatic Mouse Grab on, issue doesn't reproduce.
Oh, and indeed I have got two monitors.

**Comment by sonninnos** [comment]
I guess it can be mark solved then, since grab is required by design, and it can't really be forced by default..

**Comment by i30817** [comment]
There can still be a warning. And I think it would prevent this bug being opened over and over again (not to mention help users).
